pkg/network/usm/usm_http2_monitor_test.go

```diff
@@ -906,6 +906,84 @@ func (s *usmHTTP2Suite) TestRawTraffic() {
 				}: 5,
 			},
 		},
+		{
+			name: "Data frame header sent separately from frame payload",
+			// Testing the scenario in which the data frame header is sent separately from the frame payload.
+			messageBuilder: func() [][]byte {
+				payload := []byte("test")
+				headerFields := removeHeaderFieldByKey(testHeaders(), "content-length")
```

Comment:
document why you're removing it

nit: a better way would be modifying the value of the header to the expected size.